client/my-sites/plugins/plugins-browser/index.jsx

```diff
@@ -83,6 +88,11 @@ export class PluginsBrowser extends Component {
 
 	componentDidMount() {
 		PluginsListStore.on( 'change', this.refreshLists );
+		PluginsStore.on( 'change', this.refreshLists );
+
+		if ( ! this.isRecommendedPluginsEnabled() ) {
+			this.visibleCategories.push( 'featured' );
```

Comment:
Note that at this point the component will have been rendered already, and React by default doesn't listen for any updates done in class properties. That being said, this might be better fitting to be called before `render()` (for example in `constructor`). But that might not be reliable enough if we don't have the list of installed plugins loaded yet...

So instead I'd recommend an alternative - to not maintain a list of visible categories at all. 'd refactor this to a `getVisibleCategories()` method that would be called on every `render()` and would guarantee us to give us valid output based on the current state and props. 

What do you think?

Reply:
What @tyxla said. `visibleCategories` shouldn't be an instance variable, but the rendering should depend only on `props` and `state`. That will be a bit tricky in this convoluted component though :)

Reply:
Agreed, @tyxla and @jsnajdr - I'll go with the `getVisibleCategories` approach here.